src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -470,7 +477,7 @@ private void ProcessEvents()
                         }
 
                         uint mask = nextEvent.mask;
-                        bool addWatch = false;
+                        bool addWatch = ((mask & (uint)AddMaskFilters) != 0);
```

Comment:
Fixed. I'm keeping it as a bool rather than inlining the bit checking for readability purposes